From 9a9bd828a2da1945840003aa41e34ecd550f5ea0 Mon Sep 17 00:00:00 2001
From: Nadav Hugim <nhugim@habana.ai>
Date: Tue, 15 Feb 2022 06:39:25 +0200
Subject: [PATCH] [SW-68159] post_build: greco: add script for spi bin

Remove legacy post-image-spi script not flexible using hard coded
address and loadable address.
This new script allow to remove hard coded setting and use parameters
from fit.

Change-Id: Icbd88310c138acaf038baaa17ab2eeb6ae1d08b9
Signed-off-by: Nadav Hugim <nhugim@habana.ai>
---

diff --git a/board/habanalabs/greco/post-image-fit.sh b/board/habanalabs/greco/post-image-fit.sh
index 8951302..de815e6 100755
--- a/board/habanalabs/greco/post-image-fit.sh
+++ b/board/habanalabs/greco/post-image-fit.sh
@@ -2,6 +2,7 @@
 
 # TODO: Remove DEV_PREFIX once the HL_TARGET is renamed to greco
 DEV_PREFIX="greco"
+DEV_NAME=`echo $HL_DEV | cut -d'_' -f1`
 
 # DDR/HBM Start address as needed by Linux
 KERNEL_START_ADDR_LO=0x00000000
diff --git a/configs/greco_arm_defconfig b/configs/greco_arm_defconfig
index 6783945..c41d873 100644
--- a/configs/greco_arm_defconfig
+++ b/configs/greco_arm_defconfig
@@ -14,7 +14,7 @@
 BR2_ROOTFS_OVERLAY="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/rootfs_overlay"
 BR2_ROOTFS_POST_BUILD_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-build.sh"
 BR2_POST_IMAGE_FW_LOADER_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/agent-fw_loader.its"
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/gaudi-create-bin-file-from-fit.sh"
 BR2_LINUX_KERNEL=y
 BR2_LINUX_KERNEL_USE_ARCH_DEFAULT_CONFIG=y
 BR2_LINUX_KERNEL_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/linux_$(HL_TARGET).fragment"
diff --git a/scripts/make/hl-common/Makefile b/scripts/make/hl-common/Makefile
index 506effb..aa26b43 100644
--- a/scripts/make/hl-common/Makefile
+++ b/scripts/make/hl-common/Makefile
@@ -35,7 +35,8 @@
 
 
 HL_FAKE_SEC ?=
-HL_ARCMGMT_SRAM ?=
+HL_ARCMGMT_SRAM ?= ""
+export HL_ARCMGMT_SRAM
 HL_TARGET_ARCH := $(addsuffix %,$(HL_SINGLE_ARCH_RELEASE)) \
 		  $(addsuffix _%,$(HL_MULTI_ARCH_RELEASE))
 
